crates/uv-macros/src/lib.rs

```diff
@@ -49,3 +49,56 @@ fn impl_combine(ast: &DeriveInput) -> TokenStream {
     };
     gen.into()
 }
+
+fn get_doc_comment(attr: &Attribute) -> Option<String> {
+    if attr.path().is_ident("doc") {
+        if let syn::Meta::NameValue(meta) = &attr.meta {
+            if let syn::Expr::Lit(expr) = &meta.value {
+                if let syn::Lit::Str(str) = &expr.lit {
+                    return Some(str.value().trim().to_string());
+                }
+            }
+        }
+    }
+    None
+}
+
+#[proc_macro_attribute]
+pub fn collect_constants(_attr: TokenStream, input: TokenStream) -> TokenStream {
+    let ast = parse_macro_input!(input as ItemImpl);
+
+    let constants: Vec<_> = ast
+        .items
+        .iter()
+        .filter_map(|item| {
+            match item {
+                ImplItem::Const(item) => {
+                    let name = item.ident.to_string();
+                    let doc = item.attrs.iter().find_map(get_doc_comment).expect("Missing doc comment");
+                    Some((name, doc))
+                }
```

Comment:
```suggestion
            ImplItem::Const(item)
                if !item.attrs.iter().any(|attr| attr.path().is_ident("attr_hidden")) =>
            {
                let name = item.ident.to_string();
                let doc = item
                    .attrs
                    .iter()
                    .find_map(get_doc_comment)
                    .expect("Missing doc comment");
                Some((name, doc))
            }
```

This is likely useful to allow `#[attr_hidden]` on an env var to prevent it being rendered in the docs using below.

```rust
#[proc_macro_attribute]
pub fn attr_hidden(_attr: TokenStream, item: TokenStream) -> TokenStream {
    item
}
```

Which would allow you to do

```rust
    ...
    /// Used to set the spawning/parent interpreter when using --system in the test suite.
    #[attr_hidden]
    pub const UV_INTERNAL__PARENT_INTERPRETER: &'static str = "UV_INTERNAL__PARENT_INTERPRETER";
    ...
```

Reply:
Thanks!